protocols/request-response/src/lib.rs

```diff
@@ -181,6 +181,19 @@ pub enum OutboundFailure {
     UnsupportedProtocols,
 }
 
+impl fmt::Display for OutboundFailure {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            OutboundFailure::DialFailure => write!(f, "failed to dial the requested peer"),
+            OutboundFailure::Timeout => write!(f, "timeout while waiting for a response"),
+            OutboundFailure::ConnectionClosed => write!(f, "connection was closed before a response was received"),
+            OutboundFailure::UnsupportedProtocols => write!(f, "the remote supports none of the requested protocols")
```

Comment:
```suggestion
            OutboundFailure::DialFailure => write!(f, "Failed to dial the requested peer."),
            OutboundFailure::Timeout => write!(f, "Timeout while waiting for a response."),
            OutboundFailure::ConnectionClosed => write!(f, "Connection was closed before a response was received."),
            OutboundFailure::UnsupportedProtocols => write!(f, "The remote supports none of the requested protocols.")
```

Would be in line with most other `Display` implementations.